### PR TITLE
Replace regresssion cleanup workflow with e2e cleanup workflow, remove legacy cleanup bits [sc-51778]

### PR DESCRIPTION
**Pull request by e3b0c442**
#### What this PR does / why we need it:
* Adds the new parallelized e2e-infra-cleanup workflow which replaces the regression-test-cleanup workflow
* Removes regression-test-cleanup workflow and action as they are no longer necessary
* Address a bug in `e2e-workspace-cleanup` where a `repository_dispatch` event would not have the workspace properly populated for deletion due to differences in how the inputs are passed.

### CODE REVIEW

File: .github/workflows/e2e-infra-cleanup.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_wrapper: false
+      - name: Initialize terraform
+        run: terraform init -reconfigure
+      - name: Get automation workspaces
+        id: get-workspaces
+        run: |
+          WORKSPACES=($(terraform workspace list | grep "nickm"))
```
Comment (syntastical) [inline .github/workflows/e2e-infra-cleanup.yaml:54]: Remove the grep?

File: .github/workflows/e2e-infra-cleanup.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          repository: replicatedhq/kots-regression-automation
+          token: ${{ secrets.E2E_GH_PAT }}
+          path: automation
+          ref: main
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_wrapper: false
+      - name: Initialize terraform
+        run: terraform init -reconfigure
```
Comment (syntastical) [inline .github/workflows/e2e-infra-cleanup.yaml:50]: You shouldn't need to reconfigure here

File: .github/workflows/e2e-infra-cleanup.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        required: true
+  schedule:
+    - cron: "0 * * * *"
+
+concurrency: e2e-infra-cleanup
+
+env:
+  AWS_DEFAULT_REGION: us-east-1
+  AWS_ACCESS_KEY_ID: ${{ secrets.E2E_TESTIM_AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.E2E_TESTIM_AWS_SECRET_ACCESS_KEY }}
```
Comment (syntastical) [inline .github/workflows/e2e-infra-cleanup.yaml:25]: Are these populated for cron?

**Review by syntastical — CHANGES_REQUESTED** [review]

File: .github/workflows/e2e-infra-cleanup.yaml (hunk 1 above)
Comment (e3b0c442) [reply .github/workflows/e2e-infra-cleanup.yaml:54]: nope, but the value's wrong. Nice catch.

File: .github/workflows/e2e-infra-cleanup.yaml (hunk 2 above)
Comment (e3b0c442) [reply .github/workflows/e2e-infra-cleanup.yaml:50]: Removed.

File: .github/workflows/e2e-infra-cleanup.yaml (hunk 3 above)
Comment (e3b0c442) [reply .github/workflows/e2e-infra-cleanup.yaml:25]: The secrets context should be available for all runs. Unfortunately not a way to test it without being on the main branch. We'll need to monitor the automated cleanup and address if it's failing.

**Review by syntastical — APPROVED** [review]

**Pull request merged**